### PR TITLE
PHPC-475: toJSON and fromJSON should throw on error

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-475

### CODE REVIEW

File: tests/bson/bson-fromJSON_error-001.phpt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+}, 'MongoDB\Driver\Exception\UnexpectedValueException'), "\n";
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECT--
+OK: Got MongoDB\Driver\Exception\UnexpectedValueException
+lexical error: invalid string in json text.
+                                       foo
+                     (right here) ------^
```
Comment (derickr) [inline tests/bson/bson-fromJSON_error-001.phpt]: Does it really add the "right here" stuff? :-)

File: src/bson.c [hunk 2]
```diff
@@ -1139,7 +1145,7 @@ PHP_FUNCTION(fromJSON)
 		RETVAL_STRINGL((const char *) bson_get_data(&b), b.len, 1);
 		bson_destroy(&b);
 	} else {
-		RETURN_NULL();
+		phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "%s", error.message);
```
Comment (derickr) [inline src/bson.c]: Are you sure error.message is always populated here, and not a NULL ptr?

**Comment by derickr** [comment]
LGTM

File: tests/bson/bson-fromJSON_error-001.phpt (hunk 1 above)
Comment (jmikola) [inline tests/bson/bson-fromJSON_error-001.phpt]: Unfortunately, yes. It comes directly from yajl. We can attempt to shorten this to a one-liner at a later point.

File: src/bson.c (hunk 2 above)
Comment (jmikola) [inline src/bson.c]: Reasonably sure based on `bson_json_reader_read()`; however, it won't hurt to add a fall-back.

File: src/bson.c [hunk 3]
```diff
@@ -1114,8 +1115,13 @@ PHP_FUNCTION(toJSON)
 		RETVAL_STRINGL(str, str_len, 1);
 		bson_free(str);
 	} else {
-		RETURN_NULL();
+		phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Could not read document from BSON reader");
 	}
+
+	if (bson_reader_read(reader, &eof) || !eof) {
+		phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Reading document did not exhaust input buffer");
```
Comment (bjori) [inline src/bson.c]: This will yield in two exceptions being thrown because the former has no "return"
